modules/PROGENY.py

```diff
@@ -1,234 +1,144 @@
-                                # Append current evidence string
-                                self.evidence_strings.append(evidenceString)
-                            else:
-                                self.logger.error("%s is not found in HGNC" % gene_symbol)
-
-            self.logger.error("%s evidence parsed"%(n-1))
-            self.logger.error("%s evidence created"%len(self.evidence_strings))
-
-    def write_evidence(self, filename=Config.PROGENY_EVIDENCE_FILENAME):
-        self.logger.info("Writing PROGENY evidence strings")
-        with open(filename, 'w') as progeny_output:
-            n = 0
-            for evidence_string in self.evidence_strings:
-                n += 1
-                self.logger.info(evidence_string.disease.id[0])
-                error = evidence_string.validate(logging)
-                if error == 0:
-                    progeny_output.write(evidence_string.to_JSON(indentation=None)+"\n")
-                else:
-                    self.logger.error("REPORTING ERROR %i" %n)
-                    self.logger.error(evidence_string.to_JSON(indentation=4))
+        
+        return evidences
+    
+    def cancer2EFO(self):
+        diseaseMappingsFile = (self.spark
+                        .read.csv("resources/cancer2EFO_mappings.tsv", sep=r'\t', header=True)
+                        .select("Cancer_type_acronym", "EFO_id")
+                        .withColumnRenamed("Cancer_type_acronym", "Cancer_type"))
+
+        self.dataframe = self.dataframe.join(
+            diseaseMappingsFile,
+            on="Cancer_type",
+            how="inner"
+)
+
+        return self.dataframe
+    
+    def pathway2Reactome(self):
+        pathwayMappingsFile = (self.spark
+                        .read.csv("resources/pathway2Reactome_mappings.tsv", sep=r'\t', header=True)
+                        .withColumnRenamed("pathway", "Pathway"))
+        
+        self.dataframe = (self.dataframe
+                .join(
+                    pathwayMappingsFile,
+                    on="Pathway",
+                    how="inner"
+                )
+                .withColumn(
+                    "target",
+                    split(col("target"), ", ") 
+                )
+                .withColumn(
+                    "target",
+                    explode("target")
+                ))
+
+        return self.dataframe
+
+    @staticmethod
+    def parseEvidenceString(row):
+        try:
+            evidence = {
+                "datasourceId" : "progeny",
+                "datatypeId" : "affected_pathway",
+                "resourceScore" : row["P.Value"],
+                "pathwayName" : row["description"],
+                "pathwayId" : row["reactomeId"],
+                "targetFromSourceId" : row["target"],
+                "diseaseFromSource" : row["Cancer_type"]
+            }
+            if "EFO_id" in row:
```

Comment:
Nice, avoids issue when mapping is skipped. If you want to be extra secure you can also check that the value of the field is not `None` but it isn't really needed.